### PR TITLE
Add cleanup process and data mapping

### PR DESCRIPTION
**Pull request by NimBuzz01**

### CODE REVIEW

File: EncounterAPI/job.bal [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+// Creates a job to be executed by the scheduler.
+class Job {
+
+    *task:Job;
+    string jobIdentifier;
+
+    // Executes this function when the scheduled trigger fires.
+    public function execute() {
+        io:println("Running...");
```
Comment (Thiyan) [inline EncounterAPI/job.bal]: Please remove this log message

File: EncounterAPI/job.bal [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    }
+}
+
+function init() returns error? {
+
+    _ = check task:scheduleJobRecurByFrequency(new Job("Data clean up job"), 600);
+
+    // This init method will read some initial encounter resource from a file and initialise the internal map
+
+    io:print("Reading the encounter data from resources/data.json and initialising the in memory encounters map");
```
Comment (Thiyan) [inline EncounterAPI/job.bal]: Please change log level to debug here

**Review by Thiyan — CHANGES_REQUESTED** [review]

File: EncounterAPI/repository.bal [hunk 3]
```diff
@@ -151,23 +174,28 @@ public isolated function search(map<string[]> searchParameters) returns r4:FHIRE
                 }
             }
             encounters = filteredList;
+            if encounters.length() > offset + count {
```
Comment (Thiyan) [inline EncounterAPI/repository.bal]: I tested this, let's remove this IF condition

**Review by Thiyan — CHANGES_REQUESTED** [review]

File: EncounterAPI/repository.bal (hunk 3 above)
Comment (NimBuzz01) [reply EncounterAPI/repository.bal]: done

**Pull request merged**